sdk/rust/oak_utils/src/tests.rs

```diff
@@ -22,72 +22,143 @@ use std::fs::File;
 use std::io::Write;
 use std::path::Path;
 
-fn write_files(dir: &Path, files: &HashMap<&str, &str>) {
+fn write_files(dir: &Path, files: &HashMap<String, String>) {
     for (filename, data) in files.iter() {
         let path = dir.join(filename);
         let mut file = File::create(path).unwrap();
         file.write_all(data.as_bytes()).expect("Write error");
     }
 }
 
-fn are_equal_files(files: &[String], expected_files: &[&str]) -> bool {
-    files.len() == expected_files.len()
-        && files.iter().zip(expected_files.iter()).all(|(a, b)| a == b)
-}
-
 #[test]
 fn get_files_test() {
     let temp_files = hashmap! {
-        "1" => "string1",
-        "2" => "string2",
-        "3" => "string3",
+        "1".to_string() => "string1".to_string(),
+        "2".to_string() => "string2".to_string(),
+        "3".to_string() => "string3".to_string(),
     };
 
     let temp_dir = tempfile::tempdir().unwrap();
     write_files(temp_dir.path(), &temp_files);
 
     let files = get_files(temp_dir.path());
-    for (filename, data) in temp_files.iter() {
-        assert_eq!(
-            files.get(&String::from(*filename)),
-            Some(&String::from(*data))
-        );
-    }
+    assert_eq!(files, temp_files);
+}
+
+#[test]
+fn filediff_action_test() {
+    assert_eq!(
+        FileDiff {
+            filename: "test".to_string(),
+            old_content: Some("1".to_string()),
+            new_content: Some("2".to_string()),
+        }
+        .action(),
+        FileAction::Update {
+            new_content: "2".to_string()
+        }
+    );
+    assert_eq!(
+        FileDiff {
+            filename: "test".to_string(),
+            old_content: Some("1".to_string()),
+            new_content: Some("1".to_string()),
+        }
+        .action(),
+        FileAction::Ignore
+    );
+    assert_eq!(
+        FileDiff {
+            filename: "test".to_string(),
+            old_content: Some("1".to_string()),
+            new_content: None,
+        }
+        .action(),
+        FileAction::Remove
+    );
+    assert_eq!(
```

Comment:
Could you add a comment here that "mod.rs" is treated as an exception? Otherwise it looks odd to someone reading this test.

Reply:
Done